CPAN/EV_fstyp.patch

```diff
@@ -0,0 +1,33 @@
+--- ev.c.orig   2011-01-11 02:56:13.000000000 +0000
++++ ev.c        2021-09-03 02:17:44.734460643 +0000
+@@ -3003,19 +3003,23 @@ infy_add (EV_P_ ev_stat *w)
+     {
+       struct statfs sfs;
+
++      int len;
++      int fstyp;
++
+       /* now local changes will be tracked by inotify, but remote changes won't */
+       /* unless the filesystem is known to be local, we therefore still poll */
+       /* also do poll on <2.6.25, but with normal frequency */
+
+       if (!fs_2625)
+         w->timer.repeat = w->interval ? w->interval : DEF_STAT_INTERVAL;
+-      else if (!statfs (w->path, &sfs)
+-               && (sfs.f_type == 0x1373 /* devfs */
+-                   || sfs.f_type == 0xEF53 /* ext2/3 */
+-                   || sfs.f_type == 0x3153464a /* jfs */
+-                   || sfs.f_type == 0x52654973 /* reiser3 */
+-                   || sfs.f_type == 0x01021994 /* tempfs */
+-                   || sfs.f_type == 0x58465342 /* xfs */))
++      else if (!statfs (w->path, &sfs, len, fstyp)
++               && (sfs.f_fstyp == 0x1373 /* devfs */
++                   || sfs.f_fstyp == 0xEF53 /* ext2/3 */
```

Comment:
Do you have any information about why this change is required? This looks like a change to an underlying module. How come we have to change this patch, when we do not the module's source?

Reply:
The patch was not changed, i added a new one for solaris/illumos and its only applied for that os. Other operating systems were not affected. 

Statfs on Solaris exists for backwards compatibility, but takes four parameters. The prototype is

/usr/include/sys/statfs.h:int statfs(const char *, struct statfs *, int, int)

Reply:
Ok, got it. Could you add some hint about the OS in the patch's name? Thanks!

Reply:
changed...